### DISCUSSION TITLE
[Feature] Apply, Not Export, Sidecar Metadata

### DISCUSSION THREAD

**Discussion by tornupnegatives** [immich-app/immich#14169] [Feature Request]
### I have searched the existing feature requests to make sure this is not a duplicate request.

- [X] Yes

### The feature

I am looking for the ability to apply, not export, sidecar metadata onto images in my Immich library. The desired behavior is that when I download or export images from the web interface or mobile app, sidecar metadata is applied to the image.

This way, when downloading images to a device such as my smartphone, the timestamps will reflect metadata edits made from within Immich.

Currently, exported images have the original metadata with which the image was imported, and updated metadata is stored as sidecar files within the Immich database.

This is _not_ the same as #10592.

### Platform

- [ ] Server
- [X] Web
- [X] Mobile

**Comment by Mraedis** [comment]
I would also like this very much, and for me this can be the default. :) Something like
- Download
- Download Original
As separate buttons.

**Reply by dgcaballero** [reply]
It would be a great idea. I'm fed up with manually entering location metadata into several photos, and when I download them, that metadata doesn't appear in the downloaded photo, but it does appear in Immich. You have to fix it so that when the image is downloaded, the metadata can be seen.

**Comment by mmomjian** [comment]
Hello, the team has discussed this a few weeks back and if this is added it will be the approach in #10592. We will not edit the original file on download. Thanks

**Reply by KittKat7** [reply]
Downloading the sidecar data along with the image makes it practically very difficult to combiled the two files on mobile. For example, I can upload an image with incomplete metadata, and fix it using Immich. If I then need to share that image with someone from my phone, I am unable to download it with updated metadata.

I understand not editing the original file in storage, but why not have a second download option which creates a temporary asset for download, and merges the metadata into that? Is this something that would be considered?

**Reply by Sola85** [reply]
The recently added image editor does set precedent for altering which file is downloaded: The downloaded file includes edits from the image editor, but not metadata edits, leaving things in an awkward middle ground.

It seems to me that currently there is no way of exporting images from immich in the way they are displayed in the webapp: 
Downloading from the file system preserves metadata (via xmp sidecars) but loses image-editor edits, while downloading from the web app preserves edits but loses modified metadata.

**Reply by binnichtaktiv** [reply]
> Hello, the team has discussed this a few weeks back and if this is added it will be the approach in #10592. We will not edit the original file on download. Thanks

You don’t edit the original file in the storage / db. Only the file that is made available for download so I don’t understand where the problem is…
Extremely annoying thing right now #29581
